### PR TITLE
lightning/backend/kv: migrate test-infra to testify

### LINKED ISSUE #29979: migrate test-infra to testify for `lightning/backend/kv`

### PR DESCRIPTION
**Pull request by ArberSephirotheca**
### What problem does this PR solve?

Issue Number: close #29979

Problem Summary:
migrate test-infra to testify for lightning/backend/kv.

### What is changed and how it works?
br/pkg/lightning/backend/kv: "sql2kv_test.go" "session_test.go"

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/kv/sql2kv_test.go [hunk 1]
```diff
@@ -205,34 +207,34 @@ func (s *kvSuite) TestDecodeIndex(c *C) {
 		SQLMode:   mysql.ModeStrictAllTables,
 		Timestamp: 1234567890,
 	})
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	pairs, err := strictMode.Encode(logger, rows, 1, []int{0, 1, -1}, "1.csv", 123)
 	data := pairs.(*KvPairs)
-	c.Assert(len(data.pairs), DeepEquals, 2)
+	require.Equal(t, len(data.pairs), 2)
```
Comment (hawkingrei) [inline br/pkg/lightning/backend/kv/sql2kv_test.go:213]: use ```require.Len``` here.

File: br/pkg/lightning/backend/kv/sql2kv_test.go [hunk 2]
```diff
@@ -578,14 +578,22 @@ func (s *benchSQL2KVSuite) SetUpTest(c *C) {
...
+	s := &benchSQL2KVSuite{
+		encoder: encoder,
+		logger:  logger,
+		row:     row,
+		colPerm: colPerm,
+	}
+	return s
 }
 
-// Run `go test github.com/pingcap/tidb/br/pkg/lightning/backend -check.b -test.v` to get benchmark result.
```
Comment (hawkingrei) [inline br/pkg/lightning/backend/kv/sql2kv_test.go:584]: Can you maintain this comment and keep it right？

File: br/pkg/lightning/backend/kv/sql2kv_test.go [hunk 3]
```diff
@@ -334,86 +336,86 @@ func (s *kvSuite) TestEncodeDoubleAutoIncrement(c *C) {
 			RowID: 70,
 		},
 	}})
-	c.Assert(tbl.Allocators(encoder.(*tableKVEncoder).se).Get(autoid.AutoIncrementType).Base(), Equals, int64(70))
+	require.EqualValues(t, tbl.Allocators(encoder.(*tableKVEncoder).se).Get(autoid.AutoIncrementType).Base(), int64(70))
```
Comment (hawkingrei) [inline br/pkg/lightning/backend/kv/sql2kv_test.go:339]: use ```require.Equal```.

File: br/pkg/lightning/backend/kv/sql2kv_test.go [hunk 4]
```diff
@@ -334,86 +336,86 @@ func (s *kvSuite) TestEncodeDoubleAutoIncrement(c *C) {
...
+	require.NoError(t, err)
+	require.Equal(t, pairs, &KvPairs{pairs: []common.KvPair{
 		{
 			Key:   []uint8{0x74, 0x80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x5f, 0x72, 0xf0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x46},
 			Val:   []uint8{0x80, 0x0, 0x1, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0},
 			RowID: 70,
 		},
 	}})
-	c.Assert(tbl.Allocators(encoder.(*tableKVEncoder).se).Get(autoid.AutoRandomType).Base(), Equals, int64(70))
+	require.EqualValues(t, tbl.Allocators(encoder.(*tableKVEncoder).se).Get(autoid.AutoRandomType).Base(), int64(70))
```
Comment (hawkingrei) [inline br/pkg/lightning/backend/kv/sql2kv_test.go:376]: Ditto

File: br/pkg/lightning/backend/kv/sql2kv_test.go [hunk 5]
```diff
@@ -334,86 +336,86 @@ func (s *kvSuite) TestEncodeDoubleAutoIncrement(c *C) {
...
+		require.NoError(t, err)
 		rowID := h.IntValue()
-		c.Assert(rowID&((1<<60)-1), Equals, i)
+		require.EqualValues(t, rowID&((1<<60)-1), i)
 		keyMap[rowID>>60] = struct{}{}
 	}
-	c.Assert(len(keyMap), Equals, 8)
-	c.Assert(tbl.Allocators(encoder.(*tableKVEncoder).se).Get(autoid.RowIDAllocType).Base(), Equals, int64(32))
+	require.EqualValues(t, len(keyMap), 8)
+	require.EqualValues(t, tbl.Allocators(encoder.(*tableKVEncoder).se).Get(autoid.RowIDAllocType).Base(), int64(32))
```
Comment (hawkingrei) [inline br/pkg/lightning/backend/kv/sql2kv_test.go:415]: Ditto

File: br/pkg/lightning/backend/kv/sql2kv_test.go [hunk 6]
```diff
@@ -334,86 +336,86 @@ func (s *kvSuite) TestEncodeDoubleAutoIncrement(c *C) {
...
+	require.NoError(t, err)
+	require.Equal(t, pairs, &KvPairs{pairs: []common.KvPair{
 		{
 			Key:   []uint8{0x74, 0x80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x5f, 0x72, 0xf0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x47},
 			Val:   []uint8{0x80, 0x0, 0x1, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0},
 			RowID: 71,
 		},
 	}})
-	c.Assert(tbl.Allocators(encoder.(*tableKVEncoder).se).Get(autoid.AutoRandomType).Base(), Equals, int64(71))
+	require.EqualValues(t, tbl.Allocators(encoder.(*tableKVEncoder).se).Get(autoid.AutoRandomType).Base(), int64(71))
```
Comment (hawkingrei) [inline br/pkg/lightning/backend/kv/sql2kv_test.go:387]: Ditto.

**Comment by hawkingrei** [comment]
Generally LGTM. Thanks for your contribution!

**Comment by hawkingrei** [comment]
@tisonkun @tangenta PTAL

File: br/pkg/lightning/backend/kv/sql2kv_test.go [hunk 7]
```diff
@@ -334,86 +336,86 @@ func (s *kvSuite) TestEncodeDoubleAutoIncrement(c *C) {
...
+	require.NoError(t, err)
 	logger := log.Logger{Logger: zap.NewNop()}
 	keyMap := make(map[int64]struct{}, 16)
 	for i := int64(1); i <= 32; i++ {
 		pairs, err := encoder.Encode(logger, []types.Datum{types.NewStringDatum(fmt.Sprintf("%d", i))}, i, []int{0, -1}, "1.csv", i*32)
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 		kvs := pairs.(*KvPairs)
-		c.Assert(len(kvs.pairs), Equals, 1)
+		require.EqualValues(t, len(kvs.pairs), 1)
```
Comment (tisonkun) [inline br/pkg/lightning/backend/kv/sql2kv_test.go:407]:
```suggestion
		require.Len(t, kvs.pairs, 1)
```

ditto others.

**Comment by ArberSephirotheca** [comment]
Thanks all for taking time to review. Please points out if there are anyting more need to be improved.

**Review by tisonkun — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**